### PR TITLE
Add reset button for Payee Step on Expense Form

### PR DESCRIPTION
**Pull request by JoyLubega**
Resolve [opencollective/opencollective/issues/4840](https://github.com/opencollective/opencollective/issues/4840)

# Description
This PR adds a Reset button on the Payee step when creating an expense.

# Screenshots
Screenshot 2022-11-06 at 4 49 09 PM
Screenshot 2022-11-06 at 4 49 18 PM

### CODE REVIEW

File: components/expenses/ExpenseForm.js [hunk 1]
```diff
@@ -77,15 +77,19 @@ const msg = defineMessages({
...
   confirmCancelEditExpense: {
+    id: 'ExpenseForm.ConfirmCancelEditExpense',
     defaultMessage: 'Are you sure you want to cancel the edits?',
   },
   clearExpenseForm: {
+    id: 'ExpenseForm.ClearExpenseForm',
     defaultMessage: 'Clear Form',
   },
   confirmClearExpenseForm: {
+    id: 'ExpenseForm.ConfirmClearExpenseForm',
```
Comment (Betree) [inline components/expenses/ExpenseForm.js:92]: Let's keep these as they were, IDs are optional and adding them will invalidate all the current translations for these strings

**Comment by Betree** [comment]
@SudharakaP Do you want to review this one?

File: components/expenses/ExpenseForm.js [hunk 2]
```diff
@@ -367,6 +371,14 @@ const ExpenseFormBody = ({
     }
   }, [formPersister, dirty, values]);
 
+  const handleClearPayeeStep = () => {
+    setShowResetModal(true);
+  };
+
+  const resetDefaultStep = () => {
+    setStep(EXPENSE_FORM_STEPS.PAYEE);
+  };
```
Comment (SudharakaP) [inline components/expenses/ExpenseForm.js:380]: Minor suggestion: I am wondering the usefulness of these two functions, they are both used only ones and thus I would say maybe we can just use these functions inline within the `ExpenseFormPayeeStep`;

```
...
handleClearPayeeStep={() => setShowResetModal(true)}
...
```

**Review by SudharakaP — CHANGES_REQUESTED** [review]
Looks good. There were some minor issue I saw especially the alignment of the `back` and `next` buttons in the payee screen.

**Comment by SudharakaP** [comment]
> @SudharakaP Do you want to review this one?

Sorry, this was in my queue and somehow with other stuff I was late. 😢

**Comment by JoyLubega** [comment]
@SudharakaP i have made the changes

**Comment by JoyLubega** [comment]
@Betree  can I have another review in this

**Review by SudharakaP — APPROVED** [review]
I've tested this and all looks good. 💯 

@Betree : We are good to merge if no objections.

**Review by Betree — APPROVED** [review]
Works for me!


@SudharakaP thanks for taking care of the review

**Pull request merged**